Classes/Search.php

```diff
@@ -312,15 +312,88 @@ public function getResponseBody()
         return $this->getResponse()->response;
     }
 
+	/**
+	 * Returns a list of solr fields that could be retrieved without applying htmlspecialchars
+	 *
+	 * @return array
+	 */
+	protected function getSkipEscapeSolrFields()
+	{
+		$trustedFieldArray = array();
+		if(!isset($GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']['search.']['skipEscapeFields']) ||
+			!is_string($GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']['search.']['skipEscapeFields'])) {
+			return $trustedFieldArray;
+		}
+
+		$trustedFields = $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']['search.']['skipEscapeFields'];
+		$trustedFieldArray = GeneralUtility::trimExplode(",", $trustedFields);
+
+		return $trustedFieldArray;
+	}
+
     /**
-     * Gets the result's documents
-     *
-     * @return \Apache_Solr_Document[]
+     * @deprecated use getResultEscapedDocuments now when possible
+     * @return array
      */
-    public function getResultDocuments()
-    {
-        return $this->getResponseBody()->docs;
-    }
+	public function getResultDocuments()
+	{
+		return $this->getResponseBody()->docs;
+	}
+
+	/**
+	 * Returns all result documents but applies htmlspecialchars on all fields retrieved
+	 * from solr except the configured fields in
+	 *
+	 * plugin.tx_solr.search.skipEscapeFields
+     *
+     * @return array
+	 */
+	public function getResultDocumentsEscaped()
```

Comment:
Agree with both, but it's not what I was wondering about. Anyway, I added a counter part method so it makes more sense I hope `getResultDocumentsRaw()`